### DISCUSSION TITLE
How to increase the access token expiration time

### DISCUSSION THREAD

**Discussion by sozakir** [ory/hydra#2357] [Q&A] [answered]
I want to perform the penetration testing of my app but due to fact the access token get expired after an hour. I am not able to test it. 

After googling I came to this doc which allow to change the acess token expiration time 

https://www.ory.sh/hydra/docs/guides/token-expiration/

Using it I able to decrease the access token expiration time to 1min but I am not able to increase to more than hour eg. 72h (3 days).

How can I increase the default access token life span. I am using hydra 1.9

**Comment by vinckr** [comment, marked as answer]
Hm weird, I don't see why you wouldn't be able to increase it? 
Can you set it with an env variable? 
` export TTL_ACCESS_TOKEN=<value>`

**Reply by sozakir** [reply]
how can i check its current value in the docker container using hydra-cli. Because for checking its lifetime is not easy when value is greater than 1h.

**Reply by vinckr** [reply]
I think you can use 
`docker exec <container_id> bash -c 'echo "$TTL_ACCESS_TOKEN"'`

**Reply by sozakir** [reply]
no interactive shell present in the container therefore I am using hycli cli

**Reply by vinckr** [reply]
can you use 
[`hydra token introspect <token> `](
https://www.ory.sh/hydra/docs/cli/hydra-token-introspect)
?

**Reply by sozakir** [reply]
I have again increased the access token expiration to **48** hours and its been ~3h, **access token** is **working**. @vinckr thanks for quick reply. I really appreciate your effort. 

Earlier I have set it **72h** and Refresh token default time is also 72h. I do not know whether it is the reason. 
I have checked the regex for access token lifespan it is `^[0-9]+(ns|us|ms|s|m|h)$`. so it should work for (72h or 48h).

 I will investigate further and update here.

**Reply by vinckr** [reply]
Nice, glad you are making progress! 
Did you try the introspect command as well?
